src/fill/alloy.rs

```diff
@@ -320,16 +325,22 @@ impl Block for alloy::rpc::types::eth::Header {
             prevrandao,
             blob_excess_gas_and_price,
         } = block_env;
-        *number = self.number;
+        *number = U256::from(self.number);
         *beneficiary = self.beneficiary;
-        *timestamp = self.timestamp;
+        *timestamp = U256::from(self.timestamp);
         *gas_limit = self.gas_limit;
         *basefee = self.base_fee_per_gas.unwrap_or_default();
         *difficulty = self.difficulty;
         *prevrandao = Some(self.mix_hash);
-        *blob_excess_gas_and_price = self
-            .blob_gas_used
-            .map(|bgu| BlobExcessGasAndPrice::new(bgu, self.withdrawals_root.is_some()));
+
+        let update_fraction = if self.prague_active() {
+            revm::primitives::eip4844::BLOB_BASE_FEE_UPDATE_FRACTION_PRAGUE
+        } else {
+            revm::primitives::eip4844::BLOB_BASE_FEE_UPDATE_FRACTION_CANCUN
+        };
+
+        *blob_excess_gas_and_price =
```

Comment:
can we make the rpc header delegate to its inner consensus header? replace this whole impl with a one-liner?

Reply:
yep, https://github.com/init4tech/trevm/pull/114/commits/a0813adb734f5e2fe2192b3b8570753b1e0926e9

much nicer